app/models/concerns/host_aspects/managed_host_extensions.rb

```diff
@@ -0,0 +1,75 @@
+require 'host_aspects'
+
+module HostAspects
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      HostAspects::ManagedHostExtensions.refresh_aspect_relations(self)
+
+      has_many :host_aspects, :foreign_key => :host_id, :inverse_of => :host do
+        def [](subject)
+          where(:aspect_subject => subject).first ||
+          select { |aspect| aspect.aspect_subject == subject}.first ||
```

Comment:
Why would you need this in addition to line #12? Also, you can use #find on a collection to get the first element matching the predicate.